### DISCUSSION TITLE
Copilot C++ Stupid behavior

### DISCUSSION THREAD

**Discussion by omhland** [community/community#16081] [Copilot Conversations]
# Copilot bad suggestions

Is copilot better at making suggestion based on text that came earlier in the document? Rather than text that comes after? In this example, Copilot makes a horrible guess about the name of the C++ class members.

```cpp
//Before suggestion
struct OrderBoundary {
    -> This is where i start typing
    const float MIN;
    const float MAX;
};

//After
struct OrderBoundary {
    //This is the suggestion i got
    OrderBoundary(double lower, double upper) : lower(lower), upper(upper) {}
    float MIN;
    float MAX;
};
``` 
The struct doesn't have any members named `lower` and `upper`. It seems like more weight is put on the name of the struct than the members `MIN` and `MAX`.

**Comment by Urethramancer** [comment]
Yes, it's horrible at looking ahead sometimes. Doesn't matter what language it is - it will suggest the exact code of the function on the line after the cursor in many cases. At least it reads your whole project for context now though. Everything but the current file is considered "before", so if you can split things up very granularly, that's a way around this annoyance.
